sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -263,14 +263,18 @@ case class StaticInvoke(
           ${ev.isNull} = ${ev.value} == null;
         """
       } else {
-        val boxedResult = ctx.freshName("boxedResult")
-        s"""
-          ${CodeGenerator.boxedType(dataType)} $boxedResult = $callFunc;
-          ${ev.isNull} = $boxedResult == null;
-          if (!${ev.isNull}) {
-            ${ev.value} = $boxedResult;
-          }
-        """
+        if (method.getReturnType.isPrimitive) {
```

Comment:
Hmm, why `returnNullable` is true when return type is primitive?

Reply:
Doesn't primitive return type go the last `else` block (i.e. `returnNullable = false`)?

Reply:
I think the caller of `StaticInvoke` can set `returnNullable` to true even though the return type is primitive, so it's better to do check here. I think I can make the code simpler by reusing the else branch.

Reply:
Yea, if the caller sets `returnNullable` correctly, `StaticInvoke` won't box primitive return value. I agree, it is possible the caller wrongly set `returnNullable` to true.